lib/bundler/installer/parallel_installer.rb

```diff
@@ -0,0 +1,94 @@
+require 'bundler/worker'
+
+
+class ParallelInstaller
+
+  class SpecInstallation
+    attr_accessor :installed, :spec, :name, :post_install_message, :enqueued
+    def initialize(spec)
+      @spec, @name = spec, spec.name
+      @installed = false
+      @enqueued = false
+      @post_install_message = ""
+    end
+
+    def installed?
+      !!installed
+    end
+
+    def has_post_install_message?
+      post_install_message.empty?
+    end
+
+    def enqueued?
+      !!enqueued
+    end
+
+    def installing?
+      !installed? && enqueued?
+    end
+
+    def ready_to_install?(specs)
+      @spec.dependencies.none? do |dep|
+        next if dep.type == :development || dep.name == @name
+        specs.reject(&:installed?).map(&:name).include? dep.name
+      end
+    end
+  end
+
+  def self.call(*options)
+    new(*options).call
+  end
+
+  def self.max_threads
+    @@max_threads ||= [Bundler.settings[:jobs].to_i-1, 1].max
```

Comment:
Class variables aren't thread-safe, so this makes me nervous. Will this ever be read from inside a thread? Can it be be an ivar set during `initialize` instead?